x-pack/plugins/fleet/server/services/actions/client.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { FleetActionsClientError } from '../../../common/errors';
+
+import type { FleetActionsClientInterface, FleetAction } from './types';
+import {
+  createAction,
+  bulkCreateActions,
+  getActionsByIds,
+  getActionsWithKuery,
+  getActionResultsByIds,
+  getActionResultsWithKuery,
+} from './actions';
+
+export class FleetActionsClient implements FleetActionsClientInterface {
+  constructor(private esClient: ElasticsearchClient, private packageName: string) {
+    if (!packageName) {
+      throw new FleetActionsClientError('packageName is required');
+    }
+  }
+
+  private _verifyPackageName(action: Partial<FleetAction>) {
```

Comment:
(optional) since this is marked `private`, you really don't have to use the `_` to donate that is private. its ok with me if you leave it :)

Reply:
Yeah, agreed. Although I think `_` names are quicker to spot than `private`. I'll leave it as it is for now.